functions/post-order.ts

```diff
@@ -49,18 +47,35 @@ export async function onRequest(ctx: Context): Promise<Response> {
       throw new Error(`Given transaction has not been mined yet. Please wait for it to be mined.`);
     }
 
-    const iface = new Interface(permit2Abi);
+    let amountDaiWei;
+    let orderId;
 
-    const txParsed = iface.parseTransaction({ data: tx.data });
+    if (type === "ubiquity-dollar") {
+      const iface = new Interface(erc20Abi);
+      const txParsed = iface.parseTransaction({ data: tx.data });
+      console.log("Parsed transaction data: ", JSON.stringify(txParsed));
 
-    console.log("Parsed transaction data: ", JSON.stringify(txParsed));
+      const errorResponse = validateTransferTransaction(txParsed, txReceipt, chainId, giftCard);
+      if (errorResponse) {
+        return errorResponse;
+      }
 
-    const errorResponse = validateTransaction(txParsed, txReceipt, chainId, giftCard);
-    if (errorResponse) {
-      return errorResponse;
-    }
+      orderId = getGiftCardOrderId(txReceipt.from, txHash);
+      amountDaiWei = txParsed.args[1];
+    } else if (type === "permit") {
+      const iface = new Interface(permit2Abi);
+
+      const txParsed = iface.parseTransaction({ data: tx.data });
+      console.log("Parsed transaction data: ", JSON.stringify(txParsed));
+
+      const errorResponse = validatePermitTransaction(txParsed, txReceipt, chainId, giftCard);
+      if (errorResponse) {
+        return errorResponse;
+      }
 
-    const amountDaiWei = txParsed.args.transferDetails.requestedAmount;
+      amountDaiWei = txParsed.args.transferDetails.requestedAmount;
+      orderId = getGiftCardOrderId(txReceipt.from, txParsed.args.signature);
```

Comment:
Here `txParsed.args.signature` refers to [method signature](https://docs.ethers.org/v5/api/utils/abi/interface/#TransactionDescription) (ex: `balanceOf(address)`), not tx signature. So on redeeming permits to gift cards users will be able to redeem only once since the 2nd time the [duplicate error](https://github.com/ubiquity/pay.ubq.fi/blob/5fb3c535bbfb751020d23fa746d3dd57fb0a4f7e/functions/post-order.ts#L95) will be thrown. @EresDev Can you confirm?

Reply:
`txParsed.args.signature` is the signature of the permit and `txParsed.signature` is the method signture.

although I'm not sure if it's working correctly because when I extracted the args the `txParsed.args` was an array not an object

Reply:
> txParsed.args.signature is the signature of the permit and txParsed.signature is the method signture.

This is correct. 

> although I'm not sure if it's working correctly because when I extracted the args the txParsed.args was an array not an object

It is [Result](https://docs.ethers.org/v5/api/utils/abi/interface/#Result) type, which inherits from `Array<any>`. You can access it both via an index or a property name. From the ethers.js docs:

> A Result is an array, so each value can be accessed as a positional argument.
Additionally, if values are named, the identical object as its positional value can be accessed by its name.

Reply:
> It is [Result](https://docs.ethers.org/v5/api/utils/abi/interface/#Result) type, which inherits from `Array<any>`. You can access elements both via an index or a property name. From the ethers.js docs:
> 
> > A Result is an array, so each value can be accessed as a positional argument.
> > Additionally, if values are named, the identical object as its positional value can be accessed by its name.

that's interesting because when I tried to access it by property it didn't work